PWGHF/DataModel/CandidateReconstructionTables.h

```diff
@@ -1531,6 +1531,178 @@ DECLARE_SOA_TABLE(HfCandScMcGen, "AOD", "HFCANDSCMCGEN", //!
                   hf_cand_sigmac::FlagMcMatchGen,
                   hf_cand_sigmac::OriginMcGen);
 
+/// D*(+) --> D0 + π(+)
+namespace hf_cand_dstar
+{
+DECLARE_SOA_EXPRESSION_COLUMN(D0Px, d0px, float, 1.f * aod::hf_cand::pxProng0 + 1.f * aod::hf_cand::pxProng1);
+DECLARE_SOA_EXPRESSION_COLUMN(D0Py, d0py, float, 1.f * aod::hf_cand::pyProng0 + 1.f * aod::hf_cand::pyProng1);
+DECLARE_SOA_EXPRESSION_COLUMN(D0Pz, d0pz, float, 1.f * aod::hf_cand::pzProng0 + 1.f * aod::hf_cand::pzProng1);
+DECLARE_SOA_DYNAMIC_COLUMN(D0ImpactParameterProduct, d0impactParameterProduct,
+                           [](float dca1, float dca2) -> float { return dca1 * dca2; });
+DECLARE_SOA_DYNAMIC_COLUMN(D0M, d0m,
+                           [](float px0, float py0, float pz0, float px1, float py1, float pz1, const std::array<double, 2>& m) -> float { return RecoDecay::m(std::array{std::array{px0, py0, pz0}, std::array{px1, py1, pz1}}, m); });
+DECLARE_SOA_DYNAMIC_COLUMN(D0M2, d0m2, //!
```

Comment:
It seems to me that most of these are identical copies of the columns from the 2-prong candidate table. Can you not reuse them here?

Reply:
Earlier I did as you suggested but the names of columns in 2-prong table are deceiving us as they represent the main candidate. While our main candidate is Dstar, so to use distinguishing names for both candidates I wrote another table.

Reply:
Indeed @vkucera the point here is both technical and for the clarity of the code: this table is meant to be joinable with the main one for D* candidates (since each D* candidate has a D0 daughter) and hence for instance `Px`, `Py` and `Pz` would be the same (which is not allowed). Moreover, one should expect to get the D* momentum when calling the momentum, mass, etc getters and not those of the D0 daughter, so adding explicitly "D0" is clearer.

Reply:
Thanks for the clarification. I assumed that the reason for creating two tables was that they are supposed to be used separately. But if that is not the use case, why do you not create just one table?

Reply:
In principle after the selector there is no need anymore to consume the table for the D0, while the one for the D* would be sufficient for most of the analyses.

Reply:
It would be useful to have the analysis task included in this PR to se what is used and how. According to the version added in https://github.com/AliceO2Group/O2Physics/pull/3567/commits/6a9992a5d4699931d1718bf249e9b8cd109d3cb8 both tables are used joined together.

Reply:
I still did not look into that commit, also because it is not ready yet, but it looks to me that the join is used because the cut on the delta mass D0 is reapplied, which is not necessary to do in the task. I would prefer to disentangle the two PRs to avoid PRs with too much code to review that keeps the PR stuck for long time. Keep also in mind that we should not decide only based on the "standard" task of D*, since others will also be developed in the future (for instance for polarisation studies, correlations, Ds1+ -> D*K0S, etc)

Reply:
Fair enough. It makes sense to have two tables if you can drop one along the way. The drawback is that it makes the code more prone to errors like filling or partitioning one and not the other and there is more joining involved.